src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -2215,13 +2291,18 @@ class Simplex_tree {
     return p;
   };
 
+  //TODO: externalize this method and `decode_extended_filtration`
   /** \brief Extend filtration for computing extended persistence. 
    * This function only uses the filtration values at the 0-dimensional simplices, 
    * and computes the extended persistence diagram induced by the lower-star filtration 
    * computed with these values. 
    * \post Note that after calling this function, the filtration 
    * values are actually modified. The function `decode_extended_filtration()` 
    * retrieves the original values and outputs the extended simplex type.
+   *
+   * @warning Currently only works for @ref SimplexTreeOptions::Filtration_value which are native
+   * float types like `float` or `double`.
```

Comment:
Where does this limitation come from?

Reply:
The "native" is probably wrong. But it has to be some kind of number (i.e. multi-parameter values make no sense for the algorithm). And I excluded integers for some reasons I cannot remember. Perhaps I wanted just to exclude unsigned integers, as the sign seems to play a role. It is also not impossible that `std::numeric_limits<Filtration_value>::quiet_NaN()` played a role, as it looks quite specific (i.e. why not just put `0`, if the value does not matter and only `p.second` does in that case?), so I just assumed that an existing NaN was important.

Reply:
Integer does not work indeed, since we scale things to the interval [-2,-1]. NaN is not important; this value is not supposed to be used, and NaN can help detect a mistaken use, but if the type does not support NaN, it would be fine to store 0 or whatever else `quiet_NaN()` returns instead.
Bah, as long as this is just the documentation and not a static_assert, saying that it only works for float/double (and not any field number type) doesn't hurt much, ok.